doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+
+Checking the output
+===================
+Several output files should have been created in the run directory from that past command.
+Most important is the ``anl-afci-177.h5`` HDF5 binary database file. You can use this file
+to bring the ARMI state back to any state point from the run for analysis. 
+
+.. admonition:: Is there a general DB viewer?
+    TerraPower uses an internal HDF5 viewer called *XTVIEW* to view the state in the HDF5
+    database. At some point this tool will either be made available, or we or someone else
+    will create a plugin for a more generic visulaization tools like VisIT or Paraview.
+    For now you are stuck exploring the HDF5 output via the ARMI API. 
+
+A generic description of the outputs is provided in :doc:`/user/outputs/index`. 
+
+Adding new output
+-----------------
+Besides adding new state variables to the ARMI database, your plugins and applications may
+also find it beneficial to generate new output files, such as plots or other reports.
+Let's add automatic plots of coolant temperature at runtime in our application. 
+
+We can use the :ref:`detail assembly feature <detail-assems>` to allow users to
+choose which assembly will get printed out, since you may not want plots for every
+assembly.
+
+Open up the ``thermalSolver.py`` plugin code and add the following function to plot a
+single assembly:
+
+.. code-block:: python
+    :caption: ``myapp/thermalSolver.py``
+
+    import matplotlib.pyplot as plt
+
+    def plotOutput(a, fname):
+        """Make a temperature vs. height plot for a single assem"""
+        heights = []
+        temperatures = []
+        height = 0.0 
+
+        for b in a:
+            heights.append(height)
+            temperatures.append(b.p.THcoolantAverageT)
+            height += b.getHeight()
+
+        plt.figure()
+        plt.plot(heights, temperatures, "-o")
+        plt.title(f"Temperatures for {a}")
+        plt.xlabel("Axial height (cm)")
+        plt.ylabel("Coolant temperature (°C)")
+        plt.savefig(fname)
+        plt.close()
+
+To get the code to run on the detail assemblies, we change the ``interactEveryNode`` hook
+to call the plot function, as follows: 
+
+.. code-block:: python
+    :caption: ``myapp/thermalSolver.py``
+
+    def interactEveryNode(self, cycle=None, timeNode=None):
+        runLog.info("Computing idealized flow rate")
+        for assembly in self.r.core:
+            runThermalHydraulics(assembly)
+
+        history = self.getInterface("history")
+        for a in history.getDetailAssemblies():
+            fname = f"temps-{a.getName()}-c{self.r.p.cycle}-n{self.r.p.timeNode}.png"
+            plotOutput(a, fname)
+
+Finally, we tell ARMI which assemblies to treat as detailed by adjusting the
+``detailAssemLocationsBOL`` setting in the run directory input file.
+
+.. code-block:: yaml
+    :caption: ``runDir/anl-afci-177.yaml``
+
+    detailAssemLocationsBOL:
+      - A7001
```

Comment:
We should already by trying to wean ourselves off of detailAssemLocations, because of how overloaded they have become. This is perhaps another great opportunity for a Plugin setting

Reply:
I argued offline hard with @youngmit  on this but he did not change his mind so I will delete the entire section about plots and output.